forgerock-auth/src/main/java/org/forgerock/android/auth/callback/AbstractProtectCallback.kt

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2024 ForgeRock. All rights reserved.
+ *
+ *  This software may be modified and distributed under the terms
+ *  of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.auth.callback
+
+import org.forgerock.android.auth.Node
+import org.json.JSONException
+import org.json.JSONObject
+
+const val _ACTION = "_action"
+const val _TYPE = "_type"
+const val CLIENT_ERROR = "clientError"
+const val PING_ONE_RISK_EVALUATION_SIGNALS = "pingone_risk_evaluation_signals"
+const val PING_ONE_PROTECT = "PingOneProtect"
+const val PROTECT_INITIALIZE = "protect_initialize"
+const val PROTECT_RISK_EVALUATION = "protect_risk_evaluation"
+const val PING_ONE_PROTECT_INITIALIZE_CALLBACK = "PingOneProtectInitializeCallback"
+const val PING_ONE_PROTECT_EVALUATION_CALLBACK = "PingOneProtectEvaluationCallback"
+
+/**
+ * Abstract Protect Callback that provides the raw content of the Callback, and common methods
+ * for sub classes to access.
+ */
+abstract class AbstractProtectCallback(raw: JSONObject, index: Int) : NodeAware, AbstractCallback(raw, index) {
+
+    /**
+     * Constructor for [AbstractProtectCallback]. Capable of parsing the [MetadataCallback] used for Protect.
+     */
+    init {
```

Comment:
we can refactor something like this..not a big deal


init {
    val type = raw.optString("type")
    if (type == "MetadataCallback") {
        raw.optJSONArray("output")?.let { output ->
            output.getJSONObject(0)?.let { nameValuePair ->
                if (nameValuePair.optString("name") == "data") {
                    nameValuePair.optJSONObject("value")?.let { value ->
                        value.keys().forEach { attr ->
                            setAttribute(attr, value.get(attr))
                        }
                    }
                }
            }
        }
    }
}